lnwallet/channel.go

```diff
@@ -6358,16 +6358,22 @@ func (lc *LightningChannel) CompleteCooperativeClose(
 	return closeTx, ourBalance, nil
 }
 
-// NewAnchorResolutions returns the anchor resolutions for all currently valid
-// commitment transactions. Because we have no view on the mempool, we can only
-// blindly anchor all of these txes down.
+// NewAnchorResolutions returns a slice of anchor resolutions with a length of
+// 3, in which it has,
+//   - at index 0, the anchor for local commitment tx, or nil.
```

Comment:
Yeah the new struct is better.

I also notice that we always have at least two anchor resolutions here, one for local and the other for remote. Sometimes we would have a third for pending remote. The only reason I made the current change is that we would be able to identify which one is the anchor for local commit tx, with the assumption that **only the local commit tx needs the deadline**.  But if all anchors share the same deadline, seems like the change here is redundant...

Reply:
Yeah, I think so. Since if you want to force close the channel, you don't really care which one of the three anchors gets anchored down, as long as one does.

> I also notice that we always have at least two anchor resolutions here,

You could have a single one in the event one side has balance below the dustlimit and no HTLCs, in which case no anchor will manifest itself.